tests/components/template/test_lock.py

```diff
@@ -1,280 +1,167 @@
-                            "entity_id": "switch.test_state",
-                        },
-                        "unlock": {
-                            "service": "switch.turn_off",
-                            "entity_id": "switch.test_state",
-                        },
-                    }
-                },
-            )
+                        "service": "switch.turn_on",
+                        "entity_id": "switch.test_state",
+                    },
+                    "unlock": {
+                        "service": "switch.turn_off",
+                        "entity_id": "switch.test_state",
+                    },
+                }
+            },
+        )
 
-        self.hass.block_till_done()
-        self.hass.start()
-        self.hass.block_till_done()
+    await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
 
-        assert self.hass.states.all() == []
+    state = hass.states.get("lock.template_lock")
+    assert state.state == lock.STATE_UNLOCKED
 
-    def test_template_static(self, caplog):
-        """Test that we allow static templates."""
-        with assert_setup_component(1, "lock"):
-            assert setup.setup_component(
-                self.hass,
-                "lock",
-                {
+
+async def test_template_syntax_error(hass):
+    """Test templating syntax error."""
+    with assert_setup_component(0, lock.DOMAIN):
+        assert await setup.async_setup_component(
+            hass,
+            lock.DOMAIN,
+            {
+                "lock": {
+                    "platform": "template",
+                    "value_template": "{% if rubbish %}",
                     "lock": {
-                        "platform": "template",
-                        "value_template": "{{ 1 + 1 }}",
-                        "lock": {
-                            "service": "switch.turn_on",
-                            "entity_id": "switch.test_state",
-                        },
-                        "unlock": {
-                            "service": "switch.turn_off",
-                            "entity_id": "switch.test_state",
-                        },
-                    }
-                },
-            )
+                        "service": "switch.turn_on",
+                        "entity_id": "switch.test_state",
+                    },
+                    "unlock": {
+                        "service": "switch.turn_off",
+                        "entity_id": "switch.test_state",
+                    },
+                }
+            },
+        )
 
-        self.hass.block_till_done()
-        self.hass.start()
-        self.hass.block_till_done()
+    await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
 
-        state = self.hass.states.get("lock.template_lock")
-        assert state.state == lock.STATE_UNLOCKED
+    assert hass.states.async_all() == []
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)